web/src/bus.rs

```diff
@@ -0,0 +1,40 @@
+use crate::Application;
+
+use std::rc::Rc;
+
+#[derive(Clone)]
+pub struct Bus<Message> {
+    publish: Rc<Box<dyn Fn(Message, &mut dyn dodrio::RootRender)>>,
+}
```

Comment:
This is a composable "publisher" of messages. I am not entirely happy with the name `Bus`... Any ideas?

Reply:
I think the word for "something to pass Messages" is a *channel*.

Reply:
Yeah. I believe this is more of a message _handler_ than a channel, as it updates the application on `publish`. We will also probably need a way to defer updates to make them asynchronous.

I think we should wait untill we have more information/use cases before we find a final name.